vividus-plugin-rest-api/src/main/java/org/vividus/http/ExtendedHttpLoggingInterceptor.java

```diff
@@ -0,0 +1,130 @@
+import org.apache.hc.core5.http.HttpEntityContainer;
+import org.apache.hc.core5.http.HttpRequest;
+import org.apache.hc.core5.http.HttpRequestInterceptor;
+import org.apache.hc.core5.http.ProtocolVersion;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.apache.hc.core5.http.protocol.HttpContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.spi.LoggingEventBuilder;
+import org.vividus.http.client.HttpResponse;
+import org.vividus.http.handler.HttpResponseHandler;
+
+public class ExtendedHttpLoggingInterceptor implements HttpRequestInterceptor, HttpResponseHandler
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpRequestExecutor.class);
+
+    private static final String NEW_LINE = System.lineSeparator();
+    private static final String HEADERS_FORMAT = String.format("%nHeaders:%n{}");
+    private static final String BODY_FORMAT = String.format("%nBody:%n{}");
+
+    private static final Set<String> LOGGED_CONTENT_TYPES = Set.of(APPLICATION_JSON.getMimeType(),
+            APPLICATION_XML.getMimeType());
+
+    private final boolean extendedLogging;
+
+    public ExtendedHttpLoggingInterceptor(boolean extendedLogging)
+    {
+        this.extendedLogging = extendedLogging;
+    }
+
+    @Override
+    public void process(HttpRequest request, EntityDetails entityDetails, HttpContext context) throws IOException
+    {
+        String argBrackets = " {}";
+        LoggingEventBuilder loggingEventBuilder = LOGGER.atInfo();
+        StringBuilder loggerFormat = new StringBuilder("Request:");
+
+        ProtocolVersion protocolVersion = request.getVersion();
+        if (protocolVersion != null)
+        {
+            loggingEventBuilder = loggingEventBuilder.addArgument(protocolVersion);
+            loggerFormat.append(argBrackets);
+        }
+        loggingEventBuilder = loggingEventBuilder.addArgument(request);
+        loggerFormat.append(argBrackets);
+
+        if (extendedLogging)
+        {
+            String headersAsString = Stream.of(request.getHeaders()).map(Object::toString)
+                    .collect(Collectors.joining(NEW_LINE));
+            loggingEventBuilder = loggingEventBuilder.addArgument(headersAsString);
+            loggerFormat.append(HEADERS_FORMAT);
+
+            if (request instanceof HttpEntityContainer httpEntityContainer)
+            {
+                HttpEntity entity = httpEntityContainer.getEntity();
+                if (entity instanceof StringEntity stringEntity)
+                {
+                    String textBody = new String(stringEntity.getContent().readAllBytes(), StandardCharsets.UTF_8);
+                    loggingEventBuilder = loggingEventBuilder.addArgument(textBody);
+                    loggerFormat.append(BODY_FORMAT);
+                }
+                else if (entity != null)
+                {
+                    int bodySizeInBytes = entity.getContent().available();
```

Comment:
getContentLength() ?

Reply:
> getContentLength

Entities can be created without explicitly specifying length:
https://github.com/apache/httpcomponents-core/blob/master/httpcore5/src/main/java/org/apache/hc/core5/http/io/entity/BasicHttpEntity.java#L67-L69
So I think it's safer to take the length right from the content